### DISCUSSION TITLE
片側基盤でProMicroが反応しません

### DISCUSSION THREAD

**Discussion by rhodium173pm** [Yowkees/keyball#479] [Q&A]
![IMG_5690-1](https://github.com/Yowkees/keyball/assets/149037568/6115aafb-6758-4a8b-9ba1-7aa4999a47e7)
![IMG_5690](https://github.com/Yowkees/keyball/assets/149037568/342cc447-93e0-460d-906c-91904dd29392)

画像の基盤でProMicroとPCを接続しても、ProMicroの赤いLEDは光らずフォームウェアの書き込みも行えませんでした。
ProMicro単体での書き込みは行えて、反対の基盤でも正常に書き込みが行えたので、ProMicroに問題は無いと考えています。

基盤のショートしている箇所などを調べたいのですがどこを確認すればよいでしょうか？宜しくお願い致します。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

症状から想像すると、VCCとGND等がショートしていると思われます。
まずはProMicro裏の半田付けを疑いますが、お写真を見る限りショートしていそうな箇所は無さそうですね。
テスターをお持ちでしたら、ProMicro裏の端子で、隣とショートしている端子がないかご確認いただき、ショートしている個所は半田付けをやり直してください。

次に疑われるのは、フルカラーLEDのどれかがショートモードで壊れている可能性があります。
LEDは湿度と熱に弱く、湿度管理が甘いとけっこう壊れやすい部品になります。

この確認方法ですが、全LEDの電源ラインを回路から切り離す方法があります。
下記画像のLED CUTと表記があるパターンの赤線部分をカッターの刃の背側（写真２の黄色丸）部でカリカリと削って、左右のパターンを切り離してください。
切り離した状態でProMicroが起動するかご確認ください。
またLEDを復活したい場合は、ジャンパの半田付けの要領で切り離したパターンに半田を盛って左右を導通させます。

以上、お手数ですがご確認お願いいたします。
![61R_LEDcut](https://github.com/Yowkees/keyball/assets/84209772/1702d964-f930-480b-9177-e25e16842f49)
![cutter](https://github.com/Yowkees/keyball/assets/84209772/1e22cfd2-3951-45cf-88c9-861170276e9b)

**Comment by rhodium173pm** [comment]
かなり間が空いてしまいました、すみません

VCCとGNDがショートしており、ピンソケットの半田を全て取りきれいにした後、新しいピンソケットで半田付けをやり直しましたがショートしたままでした。
LED CUTしても変わりませんでしたが、LED CUTは私の場合、二枚目の写真の上部の箇所であっていますでしょうか？
ProMicro裏の他に確認できる箇所はありますか？

**Comment by muchikawa** [comment]
横から失礼します。
私も同じ症状でしたが、LED CUTと表記があるパターンを削って試したら、認識し動くようになりました。
結構ガリガリと削らないと駄目なようです。

ちなみに、LEDを復活させたい場合、1個づつ外して確認するしか方法はないのでしょうか？

**Comment by Yowkees** [comment]
rhodium173pm様
メッセージを見落としており、ご回答が大変遅れて申し訳ありません。
二枚目の写真の上部と、もう一か所、その裏面にもLED_CUTがありますので、この２箇所を順に試してみてください。（先に後者の裏面の可能性が高いので、こちらからやってみてください）

muchikawa様
アドバイスありがとうございます。
LEDを復活させるには、残念ながら１つずつ外して確認するしかないと思います。
非常に大変なので、もうすべてのLEDを取り外してしまい、低湿度で管理されたLEDと低温の半田ゴテでやり直した方が良いかもしれません。これらが満たされていればLEDの故障はほとんどありません。

以上、お手数ですがご対応お願いいたします。

**Reply by muchikawa** [reply]
Yowkees 様　返信ありがとうございます。
LEDの修正方法解説ありがとうございます。
全外し入れ替えやってみます。

**Comment by rhodium173pm** [comment]
Yowkees 様　
ありがとうございます
後者裏面のLED_CUTで動きました。こちらは回路図のJP19ということでしょうか？
